examples/api/test/material/bottom_app_bar/bottom_app_bar.1_test.dart

```diff
@@ -0,0 +1,56 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_api_samples/material/bottom_app_bar/bottom_app_bar.1.dart' as example;
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  testWidgets(
+      'BottomAppBarDemo shows FloatingActionButton and responds to toggle',
+      (WidgetTester tester) async {
+    await tester.pumpWidget(const example.BottomAppBarDemo());
+
+    expect(find.byType(FloatingActionButton), findsOneWidget);
+
+    // Tap the 'Floating Action Button' switch to hide the FAB.
+    await tester.tap(find.byType(SwitchListTile).first);
+    await tester.pumpAndSettle();
+
+    expect(find.byType(FloatingActionButton), findsNothing);
+  });
+
+  testWidgets('Notch can be toggled on and off', (WidgetTester tester) async {
+    await tester.pumpWidget(const example.BottomAppBarDemo());
+
+    // Check the BottomAppBar has a notch initially.
+    BottomAppBar bottomAppBar = tester.widget(find.byType(BottomAppBar));
+    expect(bottomAppBar.shape, isNotNull);
+
+    // Toggle the ['Notch'] switch to remove the notch.
+    await tester.tap(find.byType(SwitchListTile).last);
+    await tester.pump(); // Rebuild the widget after the state change
```

Comment:
```suggestion
    await tester.pump();
```
Consider removing this comment because it is very generic.